lib/src/environment/state/environment_state.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:flutter/widgets.dart';
+
+/// An observable key-value store containing values that control how Scenes are
+/// presented, such as whether dark mode is enabled, whether bold text is
+/// enabled, etc.
+class EnvironmentState extends ChangeNotifier {
+  /// Creates a new instance of [EnvironmentState] for testing purposes.
+  factory EnvironmentState.forTest() => EnvironmentState._internal();
+
+  /// Creates an [EnvironmentState] with default values for dark mode, text
+  /// scale, etc.
+  EnvironmentState._internal() {
+    _stateMapDefaults = Map<String, dynamic>.from(builtInStateDefaults);
```

Comment:
Do we need to special case the default mechanism here, or can you just set the default on the individual built-in environment control objects when you instantiate them, the same as you'd do for custom controls?

Reply:
We do not! Fixed.